### PR TITLE
setup_notification command

### LINKED ISSUE #1: Configure django rest applications using environment variables
### Description

Registration APIs (but not limited to these projects) should be configurable without manual actions in the Admin (following the configuration-as-code infrastructure principle). So the configuration using only environment variables is needed for the following APIs:
* Open Zaak (1st prio)
* Open Notificaties (1st prio)
* Objects API (2nd prio)
* Objecttypes API (2nd prio)
* Open Klant (2nd prio)

### Tasklist

**1st iteration**

* create a management command `setup_configuration`, which would run through all the configuration steps, execute them and (optionally) self-test them. This management command should be run in non-interactive mode. It can be executed in k8s job or init container. 
* The command should have an option `overwrite`, which will replace the existing configuration - making the command viable for first-time setup or to enforce a specific configuration
* the steps and their order should be set up as a django setting in the same fashion as, for example, `MIDDLEWARE` variable
* each step should have the same shape/signature and be a subclass of the same `ConfigurationStep` base class 
* each API will create its own configurations steps with provided env variables.

**2nd iteration** 

* some env variables can point to yaml files to store complicated configuration for the whole tables (for example, `zgw_consumers.Service`). The could be used to load data similar to django fixtures, but with a framework-agnostic format. Where relevant, the data in the YAML files should follow the shape/format as the API endpoints that disclose this information. The files can be mounted as files or configmaps (kubernetes) to provide them to the application container(s).
* each relevant library loading the data is responsible for dictating the format, parsing,  validating and importing the data
* loading the data needs to happen in an atomic transaction by dropping existing records and re-inserting them, or possibly updating existing records (there is a complexity risk here with breaking FK references). The truncate + load approach is preferred for simplicity sake

### Result / acceptance criteria

* a docker compose for Open Zaak and Open Notificaties, which has with init containers with `setup_configuration`command. The APIs should be able to interaction with each other and pass a smoke test 

**Related issues in other projects**

- https://github.com/maykinmedia/mozilla-django-oidc-db/issues/69

### PR DESCRIPTION
**Pull request by annashamray**
fixes #1

### CODE REVIEW

**Review by sergei-maertens — COMMENTED** [review]
I think it makes more sense to run all phases of a step and only then move to the next step, so that you can sort out dependencies between steps.

Now the order of steps is irrelevant since you first check all pre-requisites, then run all configurations and then the self tests but that breaks if you have a step that requires another step to be completed first

**Comment by annashamray** [comment]
@sergei-maertens thank you for the review!

File: django_setup_configuration/configuration.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    def validate_requirements(self) -> None:
+        """
+        check prerequisites of the configuration
+
+        :raises: :class: `django_setup_configuration.exceptions.PrerequisiteFailed`
+        if prerequisites are missing
+        """
+        missing = [
+            var for var in self.required_settings if not getattr(settings, var, None)
```
[the remark this answers is not in the document]
Comment (joeribekker) [reply django_setup_configuration/configuration.py]: I guess if it can be None, its not required?

**Review by joeribekker — APPROVED** [review]
Just needs Django 4.2 support so it won't hold us back in migrating projects.

**Comment by annashamray** [comment]
@joeribekker Django 4.2 is supported now

**Review by stevenbal — APPROVED** [review]

**Pull request merged**